posthog/models/cohort/cohort.py

```diff
@@ -103,16 +103,24 @@ def properties(self) -> PropertyGroup:
             for group in self.groups:
                 if group.get("properties"):
 
-                    # KLUDGE: map 'event' to 'person' to handle faulty event type that used to be saved
+                    # KLUDGE: map 'event' to 'person' to handle faulty event type that
+                    # used to be saved for old properties
                     # TODO: Remove once the event type is swapped over
                     props = group.get("properties")
                     if isinstance(props, list):
                         for prop in props:
-                            if prop.get("type") == "event":
+                            if prop.get("type", "event") == "event":
                                 prop["type"] = "person"
-                    elif isinstance(props, dict):
-                        if props.get("type") == "event":
-                            props["type"] = "person"
+                    elif isinstance(props, dict) and "type" not in props and "values" not in props:
```

Comment:
@EDsCODE asking for a review again because I realised that just switching the type key wouldn't work for old-old properties 😅